File: src/main/java/raccoonlang/parser/ast/DataClassBodyAstNode.java
package raccoonlang.parser.ast;

import raccoonlang.parser.Parser;
import raccoonlang.tokenizer.TokenType;

public class DataClassBodyAstNode {
    public static DataClassBodyAstNode parse(Parser parser) {
        DataClassBodyAstNode dataClassAstNode = new DataClassBodyAstNode();

        parser.take(TokenType.OPEN_CURLY);
        parser.take(TokenType.CLOSE_CURLY);

        return dataClassAstNode;
    }
}

File: Parser/AST/DataClassBodyAstNode.cs
namespace Raccoonlang.Parser.AST;

using Tokenizer;

public class DataClassBodyAstNode
{
    public List<DataClassMemberAstNode> memberList = new List<DataClassMemberAstNode>();

    public static DataClassBodyAstNode Parse(Parser parser)
    {
        DataClassBodyAstNode node = new DataClassBodyAstNode();

        parser.Take(TokenType.OPEN_CURLY);

        while(true) {
            DataClassMemberAstNode? memberNode = DataClassMemberAstNode.TryParse(parser);
            if (memberNode == null) break;
            node.memberList.Add(memberNode);
        }

        parser.Take(TokenType.CLOSE_CURLY);

        return node;
    }
}